language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ServiceDeclarationNodeContext.java

```diff
@@ -143,6 +151,13 @@ public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context,
         return completionItems;
     }
 
+    private boolean isInBasicPathContext(BallerinaCompletionContext context) {
+        return context.getNodeAtCursor().textRange().endOffset() == context.getCursorPositionInTree()
```

Comment:
Shall we use the `ServiceDeclarationNode` instead of using the `context.nodeAtCursor()`? And we should not perform string operations (split and compare) on the source code itself. Instead, we can use the `node.absoluteResourcePath()` to identify whether the cursor is inside the `context path` range.

`node.absoluteResourcePath().get(node.absoluteResourcePath().size()-1)` will give your the last part of the context path. Check if the cursor is after the end offset of that node.

Reply:
As per the offline discussion I had with @LakshanWeerasinghe, parser's behavior is incorrect around the `absoluteResourcePath` segment. Need to get that improved